openpype/hosts/resolve/api/plugin.py

```diff
@@ -391,30 +398,70 @@ def load(self, files):
         # create project bin for the media to be imported into
         self.active_bin = lib.create_bin(self.data["binPath"])
 
-        handle_start = self.data["versionData"].get("handleStart") or 0
-        handle_end = self.data["versionData"].get("handleEnd") or 0
-
+        # create mediaItem in active project bin
+        # create clip media
         media_pool_item = lib.create_media_pool_item(
             files,
             self.active_bin
         )
         _clip_property = media_pool_item.GetClipProperty
 
+        # get handles
+        handle_start = self.data["versionData"].get("handleStart")
+        handle_end = self.data["versionData"].get("handleEnd")
+        if handle_start is None:
+            handle_start = int(self.data["assetData"]["handleStart"])
+        if handle_end is None:
+            handle_end = int(self.data["assetData"]["handleEnd"])
+
+        # check frame duration from versionData or assetData
+        frame_start = self.data["versionData"].get("frameStart")
+        if frame_start is None:
+            frame_start = self.data["assetData"]["frameStart"]
+
+        # check frame duration from versionData or assetData
+        frame_end = self.data["versionData"].get("frameEnd")
+        if frame_end is None:
+            frame_end = self.data["assetData"]["frameEnd"]
+
+        db_frame_duration = int(frame_end) - int(frame_start) + 1
+
+        # get timeline in
+        timeline_start = self.active_timeline.GetStartFrame()
+        if self.sequential_load:
+            # set timeline start frame
+            timeline_in = int(timeline_start)
+        else:
+            # set timeline start frame + original clip in frame
+            timeline_in = int(
+                timeline_start + self.data["assetData"]["clipIn"])
+
         source_in = int(_clip_property("Start"))
         source_out = int(_clip_property("End"))
-
-        if _clip_property("Type") == "Video":
+        source_duration = int(_clip_property("Frames"))
+
+        # check if source duration is shorter than db frame duration
+        source_with_handles = True
+        if source_duration < db_frame_duration:
+            source_with_handles = False
+
+        # only exclude handles if source has no handles or
+        # if user wants to load without handles
+        if (
+            not self.with_handles
+            or not source_with_handles
+        ):
```

Comment:
I believe this condition should be working as substitution for the previous case where only **Video** property type was allowed `if _clip_property("Type") == "Video":`. Here we are not excluding handles to all, but only to those resources, which are either missing handles or it is chosed by user not to load with handles.

Reply:
I personally don't understand the logic.

```
        source_with_handles = True
        if source_duration < db_frame_duration:
            source_with_handles = False
```
This does not tell me whether the source has no handles, right? Could you explain your thought process?

To me it's this:
```python
start_handles = version_data.get("handleStart", 0)
end_handles = version_data.get("handleEnd", 0)
```
And **never** the asset handles on file load - the version in the database, is the ground truth of the file I'd argue.

My argument is that: If a publish has no handles data defined, then it has no handles. Again, it's an issue at the time of ingesting not something we need to solve here. As such, we don't need to somehow magically "solve" what the handles for a file are - the database is the ground truth of what we're expecting to load. If it doesn't tell me it has handles, then it has no handles.

Reply:
```
source_with_handles = True
if source_duration < db_frame_duration:
            source_with_handles = False
```
By defautl we asume source data in representation are having handles. 
Comparing length of frams in database on version data with representation amount of frames and if those are smaller then source is with no handles. 

```
start_handles = version_data.get("handleStart", 0)
end_handles = version_data.get("handleEnd", 0)
```
This solution would not work in context of this comment https://github.com/ynput/OpenPype/pull/5665#discussion_r1381523374

I am closing the conversation since we really need to mege this PR. 

Any other suggestions can be created in form of issue referencing this PR.

Reply:
Ah I see - I finally understand.

In that case your logic is wrong because now you're checking whether the duration of the file is shorter than the database frame length WITHOUT handles. You're checking it against the frame range without handles.

Instead you **should** be doing this:
```python
frame_start = version_data.get("frameStart")
frame_end = version_data.get("frameEnd")
if frame_start is not None and frame_end is not None:
    # The version has defined frame range data
    handle_start = version_data.get("handleStart", 0)
    handle_end = version_data.get("handleEnd", 0)
else:
    # Unfortunately the publish we are loading has no knowledge about its frame range
    # so all we can really do is rely on the asset's frame range as a fallback.
    # TODO: If the source does not have any frame range information we should actually
    #   be performing no custom fallbacks whatsoever
    frame_start = asset_data["frameStart"]
    frame_end = asset_data["frameEnd"]
    handle_start = asset_data["handleStart"]
    frame_end = asset_data["handleEnd"]

frame_start_handle = frame_start - handle_start
frame_end_handle = frame_end + handle_end

# The representation we are loading *can be* a representation with less frames than the version
# if it were e.g. a preview output without handles of the version. Best case scenario this data should
# be stored with the `representation` as its range explicitly, but for now we will check whether the
# loaded media is actually shorter than the database range or not - if so, we will assume the file
# is without handles
db_duration = frame_end_handle - frame_start_handle + 1
source_with_handles = source_duration >= db_duration 
```